iModelCore/ecobjects/src/ECcontext.cpp

```diff
@@ -163,10 +163,25 @@ void ECSchemaReadContext::AddSchemaPath(WCharCP path)
     SearchPathSchemaFileLocaterPtr locator = SearchPathSchemaFileLocater::CreateSearchPathSchemaFileLocater(pathVector, m_includeFilesWithNoVerExt);
 
     m_searchPaths.insert(pathStr.GetName());
-    m_locaters.insert(m_locaters.begin() + m_userAddedLocatersCount + ++m_searchPathLocatersCount, locator.get());
+    int offset = 1;
+    m_searchPathLocatersCount++;
+    if(!addOnTop)
+        offset = m_userAddedLocatersCount + m_searchPathLocatersCount;
+
+    m_locaters.insert(m_locaters.begin() + offset, locator.get());

```

Comment:
I don't see any explicit test for this new behavior

Reply:
There are no existing explicit tests for the old behavior neither. There is some stuff in SchemaDeserializationTests, none of which is about priority order. And some in SchemaLocatorTests, but also nothing about the priority order of this method.
I think we need to add comprehensive ReadContext tests, which is beyond what I want to do in this story to be honest...
Testing SchemaPath requires setting up several test directories on the local disk. Since I suggested changes to SchemaReadContext in another story, I've added a bullet "cover locater and schema path priority with explicit tests".
For now, I'll add an explicit single test for this in a new file.